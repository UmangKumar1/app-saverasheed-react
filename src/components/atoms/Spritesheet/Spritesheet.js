import React, { Component, createRef } from 'react'
import { head, map, values, flatten, prop, pipe } from 'ramda'
import { Stage, Layer, Sprite } from 'react-konva'
import PropTypes from 'prop-types'
import styled from 'react-emotion'

const { bool, shape, objectOf, number, string } = PropTypes

const StyledStage = styled(Stage)({
  pointerEvents: 'none',
})

class Spritesheet extends Component {
  constructor(props) {
    super(props)
    const { data, src } = props
    const { frames } = data
    this.state = { running: false }

    this.animations = {
      idle: pipe(
        values,
        map(
          pipe(
            prop('frame'),
            values,
          ),
        ),
        flatten,
      )(frames),
    }
    this.spriteImage = new Image()
    this.spriteImage.onload = () => this.setState({ running: true })
    this.spriteImage.src = src
  }

  static propTypes = {
    src: string.isRequired,
    data: shape({
      frames: objectOf(
        shape({
          frame: shape({
            x: number.isRequired,
            y: number.isRequired,
            w: number.isRequired,
            h: number.isRequired,
          }).isRequired,
          sourceSize: shape({
            w: number.isRequired,
            h: number.isRequired,
          }).isRequired,
        }).isRequired,
      ).isRequired,
    }).isRequired,
    stageWidth: number,
    stageHeight: number,
    autoheight: bool,
  }

  sprite = createRef()

  componentDidUpdate(prevProps, prevState) {
    const { running: wasRunning } = prevState
    const { running } = this.state
    if (running && !wasRunning) {
      if (
        this.sprite &&
        this.sprite.current &&
        !this.sprite.current.isRunning()
      ) {
        this.sprite.current.start()
      }
    }
  }

  render() {
    const { data, stageWidth, stageHeight, autoheight } = this.props
    if (autoheight) {
      return (
        <StyledStage width="auto" height="auto">
          <Layer>
            <Sprite
              ref={this.sprite}
              image={this.spriteImage}
              frameRate={30}
              animation="idle"
              x={0}
              y={0}
              animations={this.animations}
            />
          </Layer>
        </StyledStage>
      )
    }
    // pull the first frame, take its dimensions -- works so far because
    // every frame has the same dimensions. may break though

    const { w, h } = pipe(
      prop('frames'),
      values,
      head,
      prop('sourceSize'),
    )(data)

    const scale = stageWidth
      ? stageWidth / w
      : stageHeight
        ? stageHeight / h
        : 1

    return (
      <StyledStage
        width={w * scale}
        height={h * scale}
        scale={{ x: scale, y: scale }}
      >
        <Layer>
          <Sprite
            ref={this.sprite}
            image={this.spriteImage}
            frameRate={30}
            animation="idle"
            x={0}
            y={0}
            animations={this.animations}
          />
        </Layer>
      </StyledStage>
    )
  }
}

export default Spritesheet
