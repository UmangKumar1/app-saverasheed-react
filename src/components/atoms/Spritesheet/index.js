export { default, default as Spritesheet } from './Spritesheet'
