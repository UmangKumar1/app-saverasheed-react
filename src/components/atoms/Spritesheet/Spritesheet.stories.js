import React from 'react'
import { storiesOf } from '@storybook/react'
import Spritesheet from '.'
import sleepwalkerData from '../../../assets/sleepwalk'
import sleepwalkerImg from '../../../assets/sleepwalk.png'
import ufoData from '../../../assets/UFO_mapPin'
import ufoImg from '../../../assets/UFO_mapPin.png'
import ufoCheckedData from '../../../assets/UFO_mapPin_checked'
import ufoCheckedImg from '../../../assets/UFO_mapPin_checked.png'
import loadingData from '../../../assets/Loading'
import loadingImg from '../../../assets/Loading.png'
import ufoData2 from '../../../assets/UFO_mapPin_color2'
import ufoImg2 from '../../../assets/UFO_mapPin_color2.png'

storiesOf('Spritesheet', module)
  .add('sleepwalker', () => (
    <Spritesheet src={sleepwalkerImg} data={sleepwalkerData} />
  ))
  .add('ufo', () => <Spritesheet src={ufoImg} data={ufoData} />)
  .add('ufo checked', () => (
    <Spritesheet src={ufoCheckedImg} data={ufoCheckedData} />
  ))
  .add('ufo color 2', () => <Spritesheet src={ufoImg2} data={ufoData2} />)
  .add('loading', () => <Spritesheet src={loadingImg} data={loadingData} />)
  .add('loading with custom width', () => (
    <Spritesheet src={loadingImg} data={loadingData} stageWidth={400} />
  ))
  .add('loading with autoheight', () => (
    <Spritesheet src={loadingImg} data={loadingData} autoheight />
  ))
