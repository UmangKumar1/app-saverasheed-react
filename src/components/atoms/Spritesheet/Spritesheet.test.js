import React from 'react'
import { shallow } from 'enzyme'
import Spritesheet from '.'
import sleepwalkerData from '../../../assets/sleepwalk'
import sleepwalkerImg from '../../../assets/sleepwalk.png'

const wrap = (props = {}) => shallow(<Spritesheet {...props} />)

describe('Spritesheet', () => {
  it('renders children when passed in', () => {
    const wrapper = wrap({
      src: sleepwalkerImg,
      data: sleepwalkerData,
    })
    expect(wrapper).toMatchSnapshot()
  })
})
