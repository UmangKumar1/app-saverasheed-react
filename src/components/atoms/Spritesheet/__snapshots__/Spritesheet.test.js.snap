// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spritesheet renders children when passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Spritesheet
    data={
      Object {
        "frames": Object {
          "RasheeDWalk30000": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30001": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 195,
              "y": 0,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30002": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 390,
              "y": 0,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30003": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 585,
              "y": 0,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30004": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 780,
              "y": 0,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30005": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 975,
              "y": 0,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30006": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 1170,
              "y": 0,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30007": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 1365,
              "y": 0,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30008": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 1560,
              "y": 0,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30009": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 1755,
              "y": 0,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30010": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 316,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30011": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 195,
              "y": 316,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30012": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 390,
              "y": 316,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30013": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 585,
              "y": 316,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30014": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 780,
              "y": 316,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30015": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 975,
              "y": 316,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30016": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 1170,
              "y": 316,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30017": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 1365,
              "y": 316,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30018": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 1560,
              "y": 316,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30019": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 1755,
              "y": 316,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30020": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 632,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30021": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 195,
              "y": 632,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30022": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 390,
              "y": 632,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30023": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 585,
              "y": 632,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30024": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 780,
              "y": 632,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30025": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 975,
              "y": 632,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30026": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 1170,
              "y": 632,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30027": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 1365,
              "y": 632,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30028": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 1560,
              "y": 632,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30029": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 1755,
              "y": 632,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30030": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 948,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
          "RasheeDWalk30031": Object {
            "frame": Object {
              "h": 316,
              "w": 195,
              "x": 195,
              "y": 948,
            },
            "rotated": false,
            "sourceSize": Object {
              "h": 316,
              "w": 195,
            },
            "spriteSourceSize": Object {
              "h": 316,
              "w": 195,
              "x": 0,
              "y": 0,
            },
            "trimmed": false,
          },
        },
        "meta": Object {
          "app": "Adobe Animate",
          "format": "RGBA8888",
          "image": "Save Rasheed_walk2.png",
          "scale": "1",
          "size": Object {
            "h": 2048,
            "w": 2048,
          },
          "version": "18.0.2.126",
        },
      }
    }
    src="sleepwalk.png"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Layer>
        <Sprite
          animation="idle"
          animations={
            Object {
              "idle": Array [
                0,
                0,
                195,
                316,
                195,
                0,
                195,
                316,
                390,
                0,
                195,
                316,
                585,
                0,
                195,
                316,
                780,
                0,
                195,
                316,
                975,
                0,
                195,
                316,
                1170,
                0,
                195,
                316,
                1365,
                0,
                195,
                316,
                1560,
                0,
                195,
                316,
                1755,
                0,
                195,
                316,
                0,
                316,
                195,
                316,
                195,
                316,
                195,
                316,
                390,
                316,
                195,
                316,
                585,
                316,
                195,
                316,
                780,
                316,
                195,
                316,
                975,
                316,
                195,
                316,
                1170,
                316,
                195,
                316,
                1365,
                316,
                195,
                316,
                1560,
                316,
                195,
                316,
                1755,
                316,
                195,
                316,
                0,
                632,
                195,
                316,
                195,
                632,
                195,
                316,
                390,
                632,
                195,
                316,
                585,
                632,
                195,
                316,
                780,
                632,
                195,
                316,
                975,
                632,
                195,
                316,
                1170,
                632,
                195,
                316,
                1365,
                632,
                195,
                316,
                1560,
                632,
                195,
                316,
                1755,
                632,
                195,
                316,
                0,
                948,
                195,
                316,
                195,
                948,
                195,
                316,
              ],
            }
          }
          frameRate={30}
          image={
            <img
              src="sleepwalk.png"
            />
          }
          x={0}
          y={0}
        />
      </Layer>,
      "height": 316,
      "scale": Object {
        "x": 1,
        "y": 1,
      },
      "width": 195,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Sprite
          animation="idle"
          animations={
            Object {
              "idle": Array [
                0,
                0,
                195,
                316,
                195,
                0,
                195,
                316,
                390,
                0,
                195,
                316,
                585,
                0,
                195,
                316,
                780,
                0,
                195,
                316,
                975,
                0,
                195,
                316,
                1170,
                0,
                195,
                316,
                1365,
                0,
                195,
                316,
                1560,
                0,
                195,
                316,
                1755,
                0,
                195,
                316,
                0,
                316,
                195,
                316,
                195,
                316,
                195,
                316,
                390,
                316,
                195,
                316,
                585,
                316,
                195,
                316,
                780,
                316,
                195,
                316,
                975,
                316,
                195,
                316,
                1170,
                316,
                195,
                316,
                1365,
                316,
                195,
                316,
                1560,
                316,
                195,
                316,
                1755,
                316,
                195,
                316,
                0,
                632,
                195,
                316,
                195,
                632,
                195,
                316,
                390,
                632,
                195,
                316,
                585,
                632,
                195,
                316,
                780,
                632,
                195,
                316,
                975,
                632,
                195,
                316,
                1170,
                632,
                195,
                316,
                1365,
                632,
                195,
                316,
                1560,
                632,
                195,
                316,
                1755,
                632,
                195,
                316,
                0,
                948,
                195,
                316,
                195,
                948,
                195,
                316,
              ],
            }
          }
          frameRate={30}
          image={
            <img
              src="sleepwalk.png"
            />
          }
          x={0}
          y={0}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "animation": "idle",
          "animations": Object {
            "idle": Array [
              0,
              0,
              195,
              316,
              195,
              0,
              195,
              316,
              390,
              0,
              195,
              316,
              585,
              0,
              195,
              316,
              780,
              0,
              195,
              316,
              975,
              0,
              195,
              316,
              1170,
              0,
              195,
              316,
              1365,
              0,
              195,
              316,
              1560,
              0,
              195,
              316,
              1755,
              0,
              195,
              316,
              0,
              316,
              195,
              316,
              195,
              316,
              195,
              316,
              390,
              316,
              195,
              316,
              585,
              316,
              195,
              316,
              780,
              316,
              195,
              316,
              975,
              316,
              195,
              316,
              1170,
              316,
              195,
              316,
              1365,
              316,
              195,
              316,
              1560,
              316,
              195,
              316,
              1755,
              316,
              195,
              316,
              0,
              632,
              195,
              316,
              195,
              632,
              195,
              316,
              390,
              632,
              195,
              316,
              585,
              632,
              195,
              316,
              780,
              632,
              195,
              316,
              975,
              632,
              195,
              316,
              1170,
              632,
              195,
              316,
              1365,
              632,
              195,
              316,
              1560,
              632,
              195,
              316,
              1755,
              632,
              195,
              316,
              0,
              948,
              195,
              316,
              195,
              948,
              195,
              316,
            ],
          },
          "frameRate": 30,
          "image": <img
            src="sleepwalk.png"
          />,
          "x": 0,
          "y": 0,
        },
        "ref": Object {
          "current": null,
        },
        "rendered": null,
        "type": "Sprite",
      },
      "type": "Layer",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Layer>
          <Sprite
            animation="idle"
            animations={
              Object {
                "idle": Array [
                  0,
                  0,
                  195,
                  316,
                  195,
                  0,
                  195,
                  316,
                  390,
                  0,
                  195,
                  316,
                  585,
                  0,
                  195,
                  316,
                  780,
                  0,
                  195,
                  316,
                  975,
                  0,
                  195,
                  316,
                  1170,
                  0,
                  195,
                  316,
                  1365,
                  0,
                  195,
                  316,
                  1560,
                  0,
                  195,
                  316,
                  1755,
                  0,
                  195,
                  316,
                  0,
                  316,
                  195,
                  316,
                  195,
                  316,
                  195,
                  316,
                  390,
                  316,
                  195,
                  316,
                  585,
                  316,
                  195,
                  316,
                  780,
                  316,
                  195,
                  316,
                  975,
                  316,
                  195,
                  316,
                  1170,
                  316,
                  195,
                  316,
                  1365,
                  316,
                  195,
                  316,
                  1560,
                  316,
                  195,
                  316,
                  1755,
                  316,
                  195,
                  316,
                  0,
                  632,
                  195,
                  316,
                  195,
                  632,
                  195,
                  316,
                  390,
                  632,
                  195,
                  316,
                  585,
                  632,
                  195,
                  316,
                  780,
                  632,
                  195,
                  316,
                  975,
                  632,
                  195,
                  316,
                  1170,
                  632,
                  195,
                  316,
                  1365,
                  632,
                  195,
                  316,
                  1560,
                  632,
                  195,
                  316,
                  1755,
                  632,
                  195,
                  316,
                  0,
                  948,
                  195,
                  316,
                  195,
                  948,
                  195,
                  316,
                ],
              }
            }
            frameRate={30}
            image={
              <img
                src="sleepwalk.png"
              />
            }
            x={0}
            y={0}
          />
        </Layer>,
        "height": 316,
        "scale": Object {
          "x": 1,
          "y": 1,
        },
        "width": 195,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Sprite
            animation="idle"
            animations={
              Object {
                "idle": Array [
                  0,
                  0,
                  195,
                  316,
                  195,
                  0,
                  195,
                  316,
                  390,
                  0,
                  195,
                  316,
                  585,
                  0,
                  195,
                  316,
                  780,
                  0,
                  195,
                  316,
                  975,
                  0,
                  195,
                  316,
                  1170,
                  0,
                  195,
                  316,
                  1365,
                  0,
                  195,
                  316,
                  1560,
                  0,
                  195,
                  316,
                  1755,
                  0,
                  195,
                  316,
                  0,
                  316,
                  195,
                  316,
                  195,
                  316,
                  195,
                  316,
                  390,
                  316,
                  195,
                  316,
                  585,
                  316,
                  195,
                  316,
                  780,
                  316,
                  195,
                  316,
                  975,
                  316,
                  195,
                  316,
                  1170,
                  316,
                  195,
                  316,
                  1365,
                  316,
                  195,
                  316,
                  1560,
                  316,
                  195,
                  316,
                  1755,
                  316,
                  195,
                  316,
                  0,
                  632,
                  195,
                  316,
                  195,
                  632,
                  195,
                  316,
                  390,
                  632,
                  195,
                  316,
                  585,
                  632,
                  195,
                  316,
                  780,
                  632,
                  195,
                  316,
                  975,
                  632,
                  195,
                  316,
                  1170,
                  632,
                  195,
                  316,
                  1365,
                  632,
                  195,
                  316,
                  1560,
                  632,
                  195,
                  316,
                  1755,
                  632,
                  195,
                  316,
                  0,
                  948,
                  195,
                  316,
                  195,
                  948,
                  195,
                  316,
                ],
              }
            }
            frameRate={30}
            image={
              <img
                src="sleepwalk.png"
              />
            }
            x={0}
            y={0}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "animation": "idle",
            "animations": Object {
              "idle": Array [
                0,
                0,
                195,
                316,
                195,
                0,
                195,
                316,
                390,
                0,
                195,
                316,
                585,
                0,
                195,
                316,
                780,
                0,
                195,
                316,
                975,
                0,
                195,
                316,
                1170,
                0,
                195,
                316,
                1365,
                0,
                195,
                316,
                1560,
                0,
                195,
                316,
                1755,
                0,
                195,
                316,
                0,
                316,
                195,
                316,
                195,
                316,
                195,
                316,
                390,
                316,
                195,
                316,
                585,
                316,
                195,
                316,
                780,
                316,
                195,
                316,
                975,
                316,
                195,
                316,
                1170,
                316,
                195,
                316,
                1365,
                316,
                195,
                316,
                1560,
                316,
                195,
                316,
                1755,
                316,
                195,
                316,
                0,
                632,
                195,
                316,
                195,
                632,
                195,
                316,
                390,
                632,
                195,
                316,
                585,
                632,
                195,
                316,
                780,
                632,
                195,
                316,
                975,
                632,
                195,
                316,
                1170,
                632,
                195,
                316,
                1365,
                632,
                195,
                316,
                1560,
                632,
                195,
                316,
                1755,
                632,
                195,
                316,
                0,
                948,
                195,
                316,
                195,
                948,
                195,
                316,
              ],
            },
            "frameRate": 30,
            "image": <img
              src="sleepwalk.png"
            />,
            "x": 0,
            "y": 0,
          },
          "ref": Object {
            "current": null,
          },
          "rendered": null,
          "type": "Sprite",
        },
        "type": "Layer",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
